cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -839,14 +854,16 @@ public void workflowSchedules(HttpRequest request, HttpResponder responder,
    * Get schedule state.
    */
   @GET
-  @Path("/apps/{app-id}/workflows/{workflow-id}/schedules/{schedule-id}/status")
+  @Path("/apps/{app-id}/workflows/{workflow-id}/schedules/{schedule-name}/status")
```

Comment:
Are you saying that because of the way the app fabric handler is implemented, an app can only contain programs? Note that APIs should never reflect implementation. If you have to, change the implementation, not the API.